src/main/scala/mesosphere/util/state/mesos/MesosStateStore.scala

```diff
@@ -0,0 +1,49 @@
+package mesosphere.util.state.mesos
+
+import mesosphere.marathon.StoreCommandFailedException
+import mesosphere.util.BackToTheFuture.Timeout
+import mesosphere.util.ThreadPoolContext
+import mesosphere.util.state.{ PersistentEntity, PersistentStore }
+import org.apache.mesos.state.{ Variable, State }
+
+import scala.concurrent.Future
+import scala.concurrent.duration.Duration
+import scala.collection.JavaConverters._
+
+class MesosStateStore(state: State, timeout: Duration) extends PersistentStore {
+
+  implicit val timeoutDuration = Timeout(timeout)
+  implicit val ec = ThreadPoolContext.context
+  import mesosphere.util.BackToTheFuture.futureToFuture
+
+  override def load(key: ID): Future[Option[PersistentEntity]] = {
+    futureToFuture(state.fetch(key)).map { variable =>
+      if (variable.value().length > 0) Some(MesosStateEntity(key, variable)) else None
```

Comment:
I thought you were an implicit hater ... :)
I replace that with a small helper function.